src/dali/plugin/input/rest/kraken.py

```diff
@@ -0,0 +1,453 @@
+        super().__init__(account_holder, datetime(2011, 7, 27, 0, 0, 0, 0), native_fiat, thread_count)
+        self.__logger: logging.Logger = create_logger(f"{self.__EXCHANGE_NAME}")
+        self._initialize_client()
+        self._client.load_markets()
+        self._client.markets_by_id.update({'BSVUSD': {'id': 'BSVUSD', _BASE_ID: 'BSV', _BASE: 'BSV', _QUOTE: 'USD'}})
+        self.base_id_to_base: Dict[str, str] = {value[_BASE_ID]: value[_BASE] for key, value in self._client.markets_by_id.items()}
+        self.use_cache: Optional[bool] = use_cache
+
+    def exchange_name(self) -> str:
+        return self.__EXCHANGE_NAME
+
+    def plugin_name(self) -> str:
+        return self.__PLUGIN_NAME
+
+    def _initialize_client(self) -> kraken:
+        return kraken(
+            {
+                "apiKey": self.__api_key,
+                "enableRateLimit": True,
+                "secret": self.__api_secret,
+            }
+        )
+
+    @property
+    def _client(self) -> kraken:
+        super_client: Exchange = super()._client
+        if not isinstance(super_client, kraken):
+            raise RP2RuntimeError("Exchange is not instance of class kraken.")
+        return super_client
+
+    def _get_process_deposits_pagination_detail_set(self) -> Optional[AbstractPaginationDetailSet]:
+        pass
+
+    def _get_process_withdrawals_pagination_detail_set(self) -> Optional[AbstractPaginationDetailSet]:
+        pass
+
+    def _get_process_trades_pagination_detail_set(self) -> Optional[AbstractPaginationDetailSet]:
+        pass
+
+    def _process_gains(
+        self,
+        in_transactions: List[InTransaction],
+        out_transactions: List[OutTransaction],
+    ) -> None:
+        pass
+
+    def _gather_api_data(self) -> Tuple[Any, Any]:
+        loaded_cache: Tuple[Any, Any] = load_from_cache(self.__CACHE_FILE)
+        if self.use_cache and loaded_cache:
+            return loaded_cache
+
+        # get initial trade history to get count
+        index: int = 0
+        count: int = int(self._client.private_post_tradeshistory(params={_OFFSET: index})[_RESULT][_COUNT])
+        trade_history: Dict[str, Dict[str, Union[str, int, None, List[str]]]] = {}
+        while index < count:
+            trade_history.update(self._process_trade_history(index))
+            index += _TRADE_RECORD_LIMIT
+
+        # reset index and count for next API call
+        index = 0
+        count = int(self._client.private_post_ledgers(params={_OFFSET: index})[_RESULT][_COUNT])
+        ledger: Dict[str, Dict[str, Union[str, int, None, List[str]]]] = {}
+        while index < count:
+            ledger.update(self._process_ledger(index))
+            index += _TRADE_RECORD_LIMIT
+
+        result = (trade_history, ledger)
+
+        if self.use_cache:
+            save_to_cache(self.__CACHE_FILE, result)
+
+        return result
+
+    def load(self, country: AbstractCountry = None) -> List[AbstractTransaction]:
```

Comment:
Why did country get added to the `load` method signature? I'm surprised the other PR was able to get through the CI without mypy complaining of an unused variable.

Reply:
It was needed by the [RP2 input plugin](https://github.com/eprbell/dali-rp2/blob/main/src/dali/plugin/input/ods/rp2_input.py) and it could arguably be needed by other plugins too. 

Overridden methods like `load()` may trigger different mypy checks with unused parameters.